src/app/recent/[...regionCodeSlug]/page.tsx

```diff
@@ -149,20 +149,21 @@ async function BirdCard({ sighting }: { sighting: Sighting }) {
 
 export default async function RecentSightingsPage({
   params,
+  searchParams,
 }: {
   params: {
     regionCodeSlug: string[];
   };
+  searchParams: { name: string };
 }) {
   // TODO: make a function that finds a RegionCode by regionCodeSlug
   // So that we can show the region name at the top of the page
-
   const regionCode = params.regionCodeSlug.join("-").toUpperCase();
   const sightings = await getRecentSightings(regionCode);
   return (
     <div className="flex min-h-screen flex-col items-center justify-between p-4 md:p-24">
       <h1 className="mb-12 text-4xl">
-        Recent bird sightings in East Sussex{" "}
+        Recent bird sightings in {searchParams.name}{" "}
```

Comment:
Using the search params is a pretty good instinct here but I think there's some potential flaws. 

- Query params can get stripped out of a URL for a variety of reasons, what happens here if there is no query param? 
- Similarly what happens if this query param changes for some reason but the rest of the URL path doesn't?
- What if we want to show the full hierarchy name? (e.g East Sussex, England, United Kingdom)

I'm also conscious that we have a bunch of nicely structured region data already, maybe there's a way to make use of that to fetch all the possible information about a region using the region code that's part of the url path?

Reply:
Well, the other way I thought of doing this was by calling actions.tsx from the regionCodeSlug page.tsx. But I really wanted to avoid doing a search again as that felt very inefficient in comparison to just handing the information via url. I see how it's a lot more volatile to use the URL though.

Would doing a search be more efficient? I could raise a PR for that 👀

Reply:
I think having a function to get region data from the region code would be incredibly handy. 

I don't think it necessarily needs to be inefficient, we can make it run on the server from our blob of data so I think it can be pretty fast! Perhaps we can try it out (either in this PR or a follow up one) and find out!

That would mean that anything that needs region data for any reason could get it from the same function.

Reply:
I was going to make a ticket but you've done it! Ace. Yeah okay, I'll claim that in the near future and make pass no.2.
It could end up being able to get all the region data as you say.